cloud-devops: begin documenting CD (#1314)
sourcegraph/handbook: content/departments/product-engineering/engineering/process/deployments/index.md

### Images

> Each Sourcegraph service is provided as a Docker image. Every commit to `main` in [sourcegraph/sourcegraph](https://github.com/sourcegraph/sourcegraph) pushes updated Docker images for all of our services to [Docker Hub](https://hub.docker.com/u/sourcegraph/) as part of our [CI pipeline](https://buildkite.com/sourcegraph/sourcegraph) (i.e. if CI is green, then Docker images have been pushed). Images are first built as "candidate" images that are pushed to GCR to with the tag format `<commit-hash>_<build-number>_candidate`. The pipeline then runs a series of tests and checks against the images. If all pipeline steps pass the images are "promoted" and pushed to DockerHub with the tag format `<build-number>_<date>_<commit-hash>`. When a new release is cut the pipelines will build a release image with the same tag as the latest [release version](https://github.com/sourcegraph/sourcegraph/tags).

Reviewer: ```suggestion
Each Sourcegraph service is provided as a Docker image. Every commit to `main` in [sourcegraph/sourcegraph](https://github.com/sourcegraph/sourcegraph) pushes updated Docker images for all of our services to [Docker Hub](https://hub.docker.com/u/sourcegraph/) as part of our [CI pipeline](https://buildkite.com/sourcegraph/sourcegraph) (i.e. if CI is green, then Docker images have been pushed). Images are first built as "candidate" images that are pushed to GCR to with the tag format `<commit-hash>_<build-number>_candidate`. The pipeline then runs a series of tests and checks against the images. If all pipeline steps pass the images are "promoted" and pushed to DockerHub with the tag format `<build-number>_<date>_<commit-hash>`. These are used by [Sourcegraph Cloud](https://handbook.sourcegraph.com/departments/product-engineering/engineering/process/deployments/instances#sourcegraph-cloud).

When [a new semver release](https://handbook.sourcegraph.com/departments/product-engineering/engineering/process/releases) is cut the pipelines, will build a release image with the same tag as the latest [release version](https://github.com/sourcegraph/sourcegraph/tags) as well. These are used by customer deployments.
```

Reviewer: Linter might complain about absolute links, you might want to change my suggestions to relative links

For pushing custom images, refer to [building Docker images for specific branches](#building-docker-images-for-a-specific-branch).

> Images are first built as "candidate" images that are pushed to GCR to with the tag format `<commit-hash>_<build-number>_candidate`. The pipeline then runs a series of tests and checks against the images. If all pipeline steps pass the images are "promoted" and pushed to DockerHub with the tag format `<build-number>_<date>_<commit-hash>`. When a new release is cut the pipelines will build a release image with the same tag as the latest [release version](https://github.com/sourcegraph/sourcegraph/tags).

Reviewer: Merge with line 33 above? (`Each Sourcegraph service is provided as a Docker image. Every commit to `main` in...`) seems to be a bit of overlap

Author: Done!

## deploy-sourcegraph

Sourcegraph Kubernetes deployments typically start off as [deploy-sourcegraph](https://github.com/sourcegraph/deploy-sourcegraph) forks. Learn more about how we advise customers to deploy Sourcegraph in Kubernetes in our [admin installation documentation](https://docs.sourcegraph.com/admin/install/kubernetes).

There is automation in place to drive automatic updates for certain deployments from `deploy-sourcegraph`:

> - [`deploy-sourcegraph-dot-com`](https://github.com/sourcegraph/deploy-sourcegraph-dot-com) utilizies on [on-commit buildkite pipeline](https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/release/.buildkite/pipeline.yaml#L27:L33) to deploy applications automatically. On commit (either manually or by renovate), buildkite runs [`kubectl-apply-all.sh`](https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/release/kubectl-apply-all.sh) roll out the new images.

Reviewer: ```suggestion
- [`deploy-sourcegraph-dot-com`](https://github.com/sourcegraph/deploy-sourcegraph-dot-com) utilizies an [Buildkite pipeline](https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/release/.buildkite/pipeline.yaml#L27:L33) to deploy applications automatically. On commits, including those from Renovate, this pipeline runs [`kubectl-apply-all.sh`](https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/release/kubectl-apply-all.sh) roll out the new images.
```
